src/map/atcommand.c

```diff
@@ -7845,39 +7845,53 @@ ACMD(monsterignore)
 
 	return true;
 }
-/*==========================================
- * @fakename
- * => Gives your character a fake name. [Valaris]
- *------------------------------------------*/
+
+/**
+ * Temporarily changes the character's name to the specified string.
+ *
+ * @code{.herc}
+ *	@fakename {<text string>{ <options>}}
+ * @endcode
+ *
+ **/
 ACMD(fakename)
 {
-	if (!*message)
-	{
-		if (sd->fakename[0])
-		{
+	char fake_name[NAME_LENGTH] = {'\0'};
+	int options = FAKENAME_OPTION_NONE;
+
+	if (*message == '\0' || sscanf(message, "%23s %3d", fake_name, &options) < 1) {
```

Comment:
This will fail to match strings containing spaces, and incorrectly parse strings containing spaces and numbers.

For example, the following will have unexpected results:

- `@fakename Hello World` -> `fake_name`: "Hello", `options`: unmodified, `sscanf()` result: 1 ("World" discarded)
- `@fakename Hello World 1` -> `fake_name`: "Hello", `options`: unmodified, `sscanf()` result: 1 ("World 1" discarded)
- `@fakename Hello 1 World` -> `fake_name`: "Hello", `options`: 1, `sscanf()` result: 2 ("World" discarded)
- `@fakename Hello 123 1` ->  `fake_name`: "Hello", `options`: 123, `sscanf()` result: 2 ("1" discarded)

It might be easier to move the options flag as the first argument (with `sscanf(message, "%d %[^\n]", &options, fake_name)`), and handle the case where `*message != '\0'` but `sscanf() < 2`, by using `message` as is (so that the flag is optional). It would still create ambiguity for those that want to use a fakename such as "123 foo", but those could just use `@fakename 0 123 foo`.

Reply:
This won't work, since `sscanf()` would fail if only the fake name is passed because it tries to parse a number as first argument and thus returns 0.
Guess we have to work with quotation marks here...

Reply:
I was thinking of something along the lines of:

```
	char buf[NAME_LENGTH] = "";
	char *fake_name = NULL;
	int options = FAKENAME_OPTION_NONE

	if (*message == '\0') {
		/* disable fakename */
	} else {
		if (sscanf(message, "%d %23[^\n]", &options, buf, &options) == 2) // new syntax
			fake_name = buf;
		else // old syntax
			fake_name = message;

		/* enable fakename */
	}
```

Reply:
Yeah, this looks good. :+1:

Reply:
Done.